torch/csrc/lazy/core/ir.cpp

```diff
@@ -41,6 +55,19 @@ hash_t OpKind::hash() const {
   return StringHash(op.toQualString());
 }
 
+hash_t OperandHashes(const OpList& operands, const hash_t& seed, bool bakeInSizes) {
+  hash_t hash = seed;
+  for (auto& operand : operands) {
+    if (!operand) {
+      hash = HashCombine(hash, static_cast<uint64_t>(kNullOpt));
+      continue;
+    }
+    auto operand_hash = bakeInSizes ? operand.hash_with_sizes() : operand.hash_without_sizes();
```

Comment:
@shunting314 Does this conflict with our plan of removing `hash_with_sizes` concept from Base Node class?

Reply:
Partly yes. Originally this part of code is TS backend specific so I can do anything specific to TS backend. But now the code is moved to ir.cpp which is backend agnostic, so I'll have to make the change backend agnostic as well.

But I think make OpearndHashes backend agnostic make sense.

To move hash fields out of base class upon this PR, one idea I have is to change `bool bakeInSizes` to a std::function. The std::function will contain backend specific logic to calculate hash for a `operand`. Any comments about that?

Reply:
I will hold the same opinion that I hope this part of logic can remain in ts_backend because ti does not apply to xla. I think there has been some effort of only exposing the `hash` virtual API and let backend defined how it was to implement it but I didn't not follow too closely after the initial discussion.

Do you mind share what you had tried? I think this pr was drafted without that context. It is also a good place to discuss whether this `size` logic applied to MLIR backend..

Reply:
I haven't come up with a PR yet since it need wait for Will's previous merge of TS backend to PyTorch master branch, and now wait for this PR.

I think this PR can go ahead without worrying my refactoring. I'll make my change based on this PR once it's in.